homewok_20/PetrolStation.java

```diff
@@ -0,0 +1,38 @@
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class PetrolStation {
+    private int amount;
+    private final Lock lock = new ReentrantLock();
+
+    public PetrolStation(int amount) {
+        this.amount = amount;
+    }
+
+    public int getAmount() {
+        return amount;
+    }
+
+    public void setAmount(int amount) {
+        this.amount = amount;
+    }
+
+    public int doRefuel(int requestedAmount) {
+        lock.lock();
+        try {
+            if (amount < requestedAmount) {
+                throw new IllegalArgumentException("There is NOT enough fuel");
```

Comment:
Would be better not to throw an exception but inform the user via an appropriate message using a logger.